WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2339,13 +2336,14 @@ public void run() {
                     boolean isNotRestarting = mRestartEditorOption == RestartEditorOptions.NO_RESTART;
                     if ((status == PostStatus.DRAFT || status == PostStatus.PENDING) && isPublishable
                             && !hasFailedMedia() && NetworkUtils.isNetworkAvailable(getBaseContext())
-                            && isNotRestarting) {
+                            && isNotRestarting && mSite.isUsingWpComRestApi()) {
```

Comment:
I'm a bit worried about that test and the fact that `savePostOnlineAndFinishAsync` will only route to a remote-auto-save. 

Is this function called in other places for self hosted sites? Does that mean we'll never save local drafts online for self hosted sites (unless the user tap "Save" which calls `EditPostActivity.uploadPost` directly)

Reply:
>Does that mean we'll never save local drafts online for self hosted sites

You are right, we should invoke the UploadService when the user creates a new (local) draft on a self-hosted site. Fixed in 0af1803c9a11102dc4b13bd1215177a5b7449df0

Reply:
Change looks good, I did a quick check on a self hosted site and the local draft auto-upload worked fine on editor exit.